qiskit/circuit/library/arithmetic/adder/ripple_carry_adder.py

```diff
@@ -0,0 +1,116 @@
+    :math:`|a_n\rangle \otimes |a_{n-1}\rangle \ldots |a_{1}\rangle \otimes |a_{0}\rangle`
+    which denotes a quantum register prepared with the value :math:`a = 2^{0}a_{0} + 2^{1}a_{1} +
+    \ldots 2^{n}a_{n}` [2].
+
+    As an example, a ripple-carry adder circuit that performs addition on two 3-qubit sized
+    registers is as follows:
+
+    .. parsed-literal::
+
+                     ┌──────┐┌──────┐                     ┌──────┐┌──────┐
+          input_a_0: ┤0     ├┤2     ├─────────────────────┤2     ├┤0     ├
+                     │      ││      │┌──────┐     ┌──────┐│      ││      │
+          input_a_1: ┤      ├┤0     ├┤2     ├─────┤2     ├┤0     ├┤      ├
+                     │      ││      ││      │     │      ││      ││      │
+          input_a_2: ┤      ├┤  MAJ ├┤0     ├──■──┤0     ├┤  UMA ├┤      ├
+                     │      ││      ││      │  │  │      ││      ││      │
+          input_b_0: ┤1 MAJ ├┤      ├┤  MAJ ├──┼──┤  UMA ├┤      ├┤1 UMA ├
+                     │      ││      ││      │  │  │      ││      ││      │
+          input_b_1: ┤      ├┤1     ├┤      ├──┼──┤      ├┤1     ├┤      ├
+                     │      │└──────┘│      │  │  │      │└──────┘│      │
+          input_b_2: ┤      ├────────┤1     ├──┼──┤1     ├────────┤      ├
+                     │      │        └──────┘  │  └──────┘        │      │
+         carry_in_0: ┤2     ├──────────────────┼──────────────────┤2     ├
+                     └──────┘                ┌─┴─┐                └──────┘
+        carry_out_0: ────────────────────────┤ X ├────────────────────────
+                                             └───┘
+
+    Here *MAJ* and *UMA* gates correspond to the gates introduced in [1]. Note that
+    in this implementation the input register qubits are ordered as all qubits from
+    the first input register, followed by all qubits from the second input register.
+    This is different ordering as compared to Figure 4 in [1], which leads to a different
+    drawing of the circuit.
+
+    **References:**
+
+    [1] Cuccaro et al., A new quantum ripple-carry addition circuit, 2004.
+    `arXiv:quant-ph/0410184 <https://arxiv.org/pdf/quant-ph/0410184.pdf>`_
+
+    [2] Vedral et al., Quantum Networks for Elementary Arithmetic Operations, 1995.
+    `arXiv:quant-ph/9511018 <https://arxiv.org/pdf/quant-ph/9511018.pdf>`_
+    """
+
+    def __init__(self,
+                 num_state_qubits: int,
+                 name: str = 'ripple_carry_adder'
+                 ) -> None:
+        r"""Create a new ripple-carry adder circuit.
+
+                Args:
+                    num_state_qubits: The number of qubits in either input register for
+                        state :math:`|a\rangle` or :math:`|b\rangle`. The two input
+                        registers must have the same number of qubits.
+                    name: The name of the circuit object.
+                """
+        qr_a = QuantumRegister(num_state_qubits, name='input_a')
```

Comment:
Can you add a check that allows only positive integers?

Reply:
Sure, will push a check for that in a moment. Just a quick question regarding this out of my own interest. Even if I don't add a specific check in this class, passing a number of qubits that's less than 1 gives the following from the register initialization: `CircuitError: "Register size must be positive (int '-1' was provided)"`. Is it still better practice to have a specific check in each class?

Reply:
Yes, it's correct that even now an error is thrown, but the traceback ends within the call to `QuantumRegister` inside the class, which is less obvious than explicitly saying that the input argument must be larger than 0. While to us it might be obvious what caused the error, that might not be the case for others 🙂

Reply:
I think everything is clear here now, so I will mark the conversation as resolved for the time being :)